lib/lightning/webhook_auth_methods.ex

```diff
@@ -655,15 +655,52 @@ defmodule Lightning.WebhookAuthMethods do
       {:error, %Ecto.Changeset{}}
       ```
   """
-  @spec schedule_for_deletion(WebhookAuthMethod.t()) ::
+
+  @spec schedule_for_deletion(WebhookAuthMethod.t(), actor: User.t()) ::
           {:ok, WebhookAuthMethod.t()} | {:error, Ecto.Changeset.t()}
-  def schedule_for_deletion(%WebhookAuthMethod{} = webhook_auth_method) do
-    deletion_date = scheduled_deletion_date()
+  def schedule_for_deletion(%WebhookAuthMethod{} = webhook_auth_method,
+        actor: %User{} = user
+      ) do
+    # Check if the webhook_auth_method is already scheduled for deletion
+    if webhook_auth_method.scheduled_deletion do
```

Comment:
Ben when you come back to this PR, appending the reasoning from our convo, it would be an idiom to program by contract, making it clear what are the pre-conditions to call the function and the scenario the function is handling. But we can have a general conversion about idioms later as well so don't feel pressured with this item, no comments actually including this one (: